node/main.go

```diff
@@ -123,7 +125,74 @@ func (s server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (s server) MinerInfo() ([]api.MinerInfo, error) {
+func (s *server) runWorker() error {
+	ctx := context.Background()
+
+	messagePoolUpdateChannel, err := s.rpc.MpoolSub(ctx)
+	if err != nil {
+		return fmt.Errorf("Unable to subscribe to message pool: %s", err)
+	}
+
+	chainNotify, err := s.rpc.ChainNotify(ctx)
+	if err != nil {
+		return fmt.Errorf("Unable to subscribe to chain notify channel: %s", err)
+	}
+
+	for {
+		select {
+		case update := <-messagePoolUpdateChannel:
+			msg := update.Message.VMMessage()
+
+			// for now, just log the details of the update
+			// in the future, use the address to identify if it's a miner or what
+			log.Info().
+				Str("to", msg.To.String()).
+				Str("from", msg.From.String()).
+				Str("value", msg.Value.String()).
+				Str("method", msg.Method.String()).
+				Msg("received update message")
+
+		case chainUpdate := <-chainNotify:
+			log.Info().Msg("received chain notification")
+
+			for _, headChange := range chainUpdate {
+				if headChange != nil {
+					log.Info().
+						Str("type", headChange.Type).
+						Str("height", headChange.Val.Height().String()).
+						Send()
+
+					go s.ProcessTipSet(headChange.Val)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *server) ProcessTipSet(ts *types.TipSet) error {
+	addrs, err := s.rpc.StateListActors(context.Background(), ts.Key())
+	if err != nil {
+		return fmt.Errorf("unable to list actors: %s", err)
+	}
+
+	log.Info().Int("number of actors", len(addrs)).Send()
+
+	// NOTE: this is too much information
```

Comment:
What did you mean here?

Reply:
Since it's like 120k addrs, it's just too much to log haha. Maybe could start processing in the background, but unsure how useful that would be.